R/transform-code.R

```diff
@@ -60,16 +98,16 @@ separate_chunks <- function(lines) {
   lst(r_chunks, text_chunks)
 }
 
-#' Identifies raw R code chunks
+#' Identifies raw Rmd code chunks
 #'
 #' Raw in the sense that these chunks don't contain pure R code, but they
 #' contain a header and footer of markdown. Only code chunks that have an engine
 #' whose name matches `engine-pattern` are considered as R code.
 #' @inheritParams separate_chunks
 #' @param engine_pattern A regular expression that must match the engine name.
 #' @keywords internal
-identify_r_raw_chunks <- function(lines, engine_pattern = get_engine_pattern()) {
```

Comment:
For some reason we did not name them *rmd chunks*, I think because of ambiguity with other functions, but I can't exactly remember why and looking at all function names that match `rmd`
```r
# remotes::install_github("lorenzwalther/parsesum")
parsesum::ls_fun_decs() %>% dplyr::filter(grepl("rmd", .$text))
#> # A tibble: 2 x 4
#>   text          n_lines source               line1
#>   <chr>           <int> <chr>                <int>
#> 1 transform_rmd       7 ./R/transform-code.R    33
#> 2 is_rmd_file         3 ./R/utils.R             83
```
Or `identify`: 
```r
parsesum::ls_fun_decs() %>% dplyr::filter(grepl("identify", .$text))
#> # A tibble: 4 x 4
#>   text                                                 n_lines source                      line1
#>   <chr>                                                  <int> <chr>                       <int>
#> 1 identify_insufficiently_parsed_stings                      8 ./R/parse.R                   198
#> 2 identify_start_to_stop_of_roxygen_examples_from_text       6 ./R/roxygen-examples-find.R     9
#> 3 identify_start_to_stop_of_roxygen_examples                 4 ./R/roxygen-examples-find.R    16
#> 4 identify_r_raw_chunks                                     20 ./R/transform-code.R           71
```

I don't find a compelling reason anymore, so let's rename this functionality.

Reply:
Yeah I did wonder about this. Figured some function names just carried through from earlier implementations.  I've now merged the `Rmd` and `Rnw` cases into a single function to eliminate some of the code duplication, and have renamed this to `identify_raw_chunks()`. I think we were using 'raw' to identify the fact that the line references here encapsulate the whole R chunk as opposed to just the lines with code that get extracted by `separate_chunks()`. Happy to change this to something else if you want to though.

Reply:
Ok cool.